### PR TITLE
Allow reading of version information from env-vars

### PR DESCRIPTION
**Pull request by MKodde**
At this point,  you can set three env-vars that are evaluated to display on the /info endpoint.

If these variables are not present, the default behavior triggers (determining the version info based on the path information.

1. By default this is docker friendly (as our OpenContext images ship with these env-vars)
2. The fallback is usefull for when you install the application on bare metal, and unzip our openconext tarbals.

### CODE REVIEW

File: src/DependencyInjection/Compiler/HealthCheckPass.php [hunk 1]
```diff
@@ -38,7 +38,7 @@ public function process(ContainerBuilder $container)
 
         $definition = $container->findDefinition('openconext.monitor.health_check_chain');
 
-        // find all service IDs with the app.mail_transport tag
+        // find all service IDs with the openconext.monitor.health_check tag
```
Comment (MKodde) [inline src/DependencyInjection/Compiler/HealthCheckPass.php:41]: Boyscout rule :evergreen_tree:

File: phpmd.xml [hunk 2]
```diff
@@ -30,6 +30,8 @@
     <rule ref="rulesets/naming.xml">
         <exclude name="ShortVariable" />
         <exclude name="LongVariable" />
+        <exclude name="LongClassName" />
+        <exclude name="ShortClassName" />
```
Comment (MKodde) [inline phpmd.xml:34]: Conform to the latest QA tool PHPMD ruleset we use in other OpenConext apps

File: composer.json [hunk 3]
```diff
@@ -37,11 +37,11 @@
   "scripts": {
     "tests": {
       "docheader": "vendor/bin/docheader check src/",
-      "phpcmd": "vendor/bin/phpmd src text phpmd.xml",
       "phpcs": "vendor/bin/phpcs src --report=full --standard=phpcs.xml --extensions=php --warning-severity=0",
       "phpcpd": "vendor/bin/phpcpd src --exclude=src/Tests/*",
-      "phpunit": "vendor/bin/phpunit --coverage-text"
-      },
+      "phpunit": "vendor/bin/phpunit --coverage-text",
+      "phpmd": "vendor/bin/phpmd src text phpmd.xml --exclude 'src/Tests/*'"
```
Comment (MKodde) [inline composer.json:43]: Exclude the Unit tests from php md sniffing.

File: src/Value/BuildInformation.php [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace OpenConext\MonitorBundle\Value;
+
+use JsonSerializable;
+
+interface BuildInformation extends JsonSerializable
```
Comment (parijke) [inline src/Value/BuildInformation.php:23]: suffix Interface or not?

File: src/Value/BuildPath.php [hunk 5]
```diff
@@ -26,45 +26,27 @@
...
-    private $path;
-
-    private $version;
-
-    private $revision;
-
-    public function __construct($path, $version = '', $revision = '')
+    public function __construct(
+        private readonly string $path,
+        private readonly string $version = '',
```
Comment (parijke) [inline src/Value/BuildPath.php]: Why not initialise all vars?

File: src/Value/Information.php [hunk 6]
```diff
@@ -26,48 +26,31 @@
...
-    /**
-     * @var array
-     */
-    private $systemInfo;
-    
-  
-    public static function buildFrom(BuildPath $buildPath, string $environment, bool $debuggerEnabled, array $systemInfo): Information
-    {
+    public static function buildFrom(
+        BuildInformation $buildPath,
```
Comment (parijke) [inline src/Value/Information.php:30]: Beware of not initialising vars (see my article as well)

**Review by parijke — CHANGES_REQUESTED** [review]

File: src/Value/BuildInformation.php (hunk 4 above)
Comment (MKodde) [reply src/Value/BuildInformation.php:23]: I'm not going to add the suffix here.

File: src/Value/BuildPath.php (hunk 5 above)
Comment (MKodde) [reply src/Value/BuildPath.php]: The path is the one param that MUST be passed to the constructor. That's why I'm not alllowing a default value.

File: src/Value/Information.php (hunk 6 above)
Comment (MKodde) [reply src/Value/Information.php:30]: I only want to set default values when they are usefull. In this case, all vars are initialized during construction time. None of the params are optional. So this should never lead to problems right?

**Pull request merged**